### PR TITLE
Configure an access log so that we can build metrics and alert on it.

### PR DESCRIPTION
**Pull request by ndmckinley**
### Description
This prints out the path, method, timing, and response code for every request, then pipes it into the STDOUT logs, then hooks those logs up to awslogs.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: universal-application-tool-0.0.1/app/filters/LoggingFilter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return EssentialAction.of(
+        request -> {
+          final long startTime = clock.millis();
+          return next.apply(request)
+              .map(
+                  result -> {
+                    long time = clock.millis() - startTime;
+                    log.info(
+                        "{}\t{}\t{}ms\t{}", request.method(), request.uri(), time, result.status());
+                    return result.withHeader("X-RequestTime", String.valueOf(time));
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/filters/LoggingFilter.java:38]: **Question**: Is there a specific string format or a predefined list of strings this (`X-RequestTime`) should follow? Or can this be an arbitrary string, i.e. it is not defined or used anywhere else?

File: universal-application-tool-0.0.1/app/filters/LoggingFilter.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import play.mvc.EssentialAction;
+import play.mvc.EssentialFilter;
+
+/** This is a simple filter that produces an apache-style access log. */
+@Singleton
+public class LoggingFilter extends EssentialFilter {
+
+  private final Executor exec;
+  private final Clock clock;
+  private static final Logger log = LoggerFactory.getLogger("loggingfilter");
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/filters/LoggingFilter.java:18]: **Question**: Is this (`loggingfilter`) a specific name (name of a library or a component)? Or is it defined by you in `logback.xml`?

**Review by hmhsu — APPROVED** [review]
I am not fully sure of the details but looks good to me!

File: infra/containers.yaml [hunk 3]
```diff
@@ -101,4 +106,10 @@ Resources:
           PortMappings:
             - ContainerPort: '9000'
               HostPort: '9000'
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-group: !Ref 'loggroup'
+              awslogs-region: us-west-2
+              awslogs-stream-prefix: uatsrv
```
[the remark this answers is not in the document]
Comment (ndmckinley) [reply infra/containers.yaml:114]: Nope, this config is the whole thing.  Just short for "uat server".

File: universal-application-tool-0.0.1/app/filters/LoggingFilter.java (hunk 2 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/filters/LoggingFilter.java:18]: It's a reference to `logback.xml`, yep!  Any string is acceptable, logback configures where the logs go by name.

File: universal-application-tool-0.0.1/app/filters/LoggingFilter.java (hunk 1 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/filters/LoggingFilter.java]: Yep, HTTP headers can be anything.  Traditionally nonstandard headers are prefixed with `X-`.

Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/filters/LoggingFilter.java]: Whoops - I looked it up and actually this convention was recommended until the mid 2010s and now it's not recommended.  We still use it at Google sometimes.  I'll remove it!

**Pull request merged**